File: service/src/main/java/com/example/chatapp/model/GroupDTO.java
package com.example.chatapp.model;

import com.example.chatapp.db.entity.MemberInvitationStatus;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class GroupDTO {
    Long id;
    String name;
    List<MemberDTO> memberDTOList;
    MemberInvitationStatus invitationStatus;
}

File: ui/src/app/chat/contacts/model/group-dto.ts
import {MemberDto} from "./member-dto";
import {MemberInvitationStatus} from "../../../service/model/member-invitation-status";

export type GroupDto = {
    id: number;
    username: number;
    memberDTOList: MemberDto[]
    adminMemberDTOList: MemberDto[]
    invitationStatus: MemberInvitationStatus
}
